packages/web/src/components/buy-sell-modal/components/CurrentWalletBanner.tsx

```diff
@@ -0,0 +1,208 @@
+import {
+  useCoinBalance,
+  useCurrentAccountUser,
+  useExternalWalletBalance
+} from '@audius/common/api'
+import { formatCurrency, shortenSPLAddress } from '@audius/common/utils'
+import {
+  Flex,
+  IconAudiusLogoColor,
+  IconPhantom,
+  IconMetamask,
+  IconSolana,
+  LoadingSpinner,
+  Skeleton,
+  Text,
+  TextLink
+} from '@audius/harmony'
+
+import { appkitModal } from 'app/ReownAppKitModal'
+import { useConnectExternalWallets } from 'hooks/useConnectExternalWallets'
+import { env } from 'services/env'
+
+const messages = {
+  tradeWith: 'Trade with',
+  builtInWallet: 'Built-in Wallet',
+  builtInWalletNotAvailable: 'Built-in wallet not available',
+  connect: 'Connect',
+  disconnect: 'Disconnect',
+  available: 'Available'
+}
+
+type WalletIconComponent =
+  | typeof IconPhantom
+  | typeof IconMetamask
+  | typeof IconSolana
+
+const getWalletIcon = (): WalletIconComponent => {
+  // Check if Phantom is available
+  if (typeof window !== 'undefined' && window.phantom) {
+    return IconPhantom
+  }
+  if (typeof window !== 'undefined' && window.ethereum) {
+    return IconMetamask
+  }
+  return IconSolana // Fallback
+}
+
+export const CurrentWalletBanner = ({
+  inputToken
+}: {
+  inputToken: { mint: string; symbol: string }
+}) => {
+  const { data: currentUser } = useCurrentAccountUser()
+  const isAnonymousUser = !currentUser
```

Comment:
nit: `isLoggedOutUser` feels more clear to me or `isUnauthed` tbh not bad to just use `!currentUser` everywhere `isAnonymousUser` is used. probably the most clear and is less chars haha